src/app/admin-page/admin-page.component.ts

```diff
@@ -1,20 +1,107 @@
 import { Component, OnInit } from '@angular/core';
+import { MatTableDataSource } from '@angular/material/table';
+import { Post } from '../model-service/post/post';
+import { Comment } from '../model-service/comment/comment';
+import { PostService } from '../model-service/post/post.service';
+import { CommentService } from '../model-service/comment/comment.service';
 
 @Component({
   selector: 'app-admin-page',
   templateUrl: './admin-page.component.html',
-  styleUrls: ['./admin-page.component.scss']
+  styleUrls: ['./admin-page.component.scss'],
 })
 export class AdminPageComponent implements OnInit {
-
   /**
    * Intended method: Include two lists/tables (your choice), one is for unapproved posts, the other is for
    * unapproved comments. Admin should be able to approve a post in minimal moves, e.g. within 2 clicks.
    */
+  posts = new MatTableDataSource<Post>();
+  comments = new MatTableDataSource<Comment>();
```

Comment:
array is sufficient here, as the table does also receive arrays as data sources. in fact as the data from the backend will be paginated, there isn't much reason to use MatTableDataSource also